src/main/java/core/mate/academy/service/MachineProducer.java

```diff
@@ -2,10 +2,10 @@
 
 import java.util.List;
 
-public interface MachineProducer {
+public interface MachineProducer<T> {
```

Comment:
```suggestion
public interface MachineProducer<T extends Machine> {
```
that will forbid such cases: `BulldozerProducer implements MachineProducer<String>`